backend/src/routes/api/userPatch.routes.js

```diff
@@ -0,0 +1,44 @@
+import { StatusCodes } from 'http-status-codes';
+import User, { userValidator } from '../../models/User.js';
+
+const userPatchEndpoint = (router) => {
+  router.patch('/users/:id', getUser, userValidator, async (req, res) => {
+    res.user_patch = {};
+    if (req.body.username != null) {
+      res.user_patch.username = req.body.username;
+    }
+    if (req.body.email != null) {
+      res.user_patch.email = req.body.email;
+    }
+    if (req.body.password != null) {
+      res.user_patch.password = req.body.password;
+    }
+    try {
+      const updatedUser = await User.updateOne(
+        { _id: req.params.id },
+        { $set: res.user_patch },
+        { upsert: true, runValidators: true },
+      );
+      res.status(StatusCodes.OK).json(updatedUser);
+    } catch (err) {
+      res.status(StatusCodes.BAD_REQUEST).json({ message: err.message });
+    }
```

Comment:
First of all, we want to extract this middleware to a separate function, so our route definition is cleaner.

Reply:
done